core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -49,34 +49,18 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   Impl::HIPInternal::m_hipDev = hip_device_id;
   KOKKOS_IMPL_HIP_SAFE_CALL(
       hipGetDeviceProperties(&Impl::HIPInternal::m_deviceProp, hip_device_id));
-  const auto& hipProp = Impl::HIPInternal::m_deviceProp;
   KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(hip_device_id));
 
-  // number of multiprocessors
-  Impl::HIPInternal::m_multiProcCount = hipProp.multiProcessorCount;
-
-  //----------------------------------
-  // Maximum number of warps,
-  // at most one warp per thread in a warp for reduction.
-  Impl::HIPInternal::m_maxWarpCount =
-      hipProp.maxThreadsPerBlock / Impl::HIPTraits::WarpSize;
-  if (Impl::HIPTraits::WarpSize < Impl::HIPInternal::m_maxWarpCount) {
-    Impl::HIPInternal::m_maxWarpCount = Impl::HIPTraits::WarpSize;
-  }
-
-  //----------------------------------
-  // Maximum number of blocks
-  Impl::HIPInternal::m_maxBlock[0] = hipProp.maxGridSize[0];
-  Impl::HIPInternal::m_maxBlock[1] = hipProp.maxGridSize[1];
-  Impl::HIPInternal::m_maxBlock[2] = hipProp.maxGridSize[2];
-
   // theoretically, we can get 40 WF's / CU, but only can sustain 32 see
-  // https://github.com/ROCm-Developer-Tools/HIP/blob/a0b5dfd625d99af7e288629747b40dd057183173/vdi/hip_platform.cpp#L742
-  Impl::HIPInternal::m_maxWavesPerCU = 32;
-  Impl::HIPInternal::m_shmemPerSM    = hipProp.maxSharedMemoryPerMultiProcessor;
-  Impl::HIPInternal::m_maxShmemPerBlock = hipProp.sharedMemPerBlock;
+  // https://github.com/ROCm/clr/blob/4d0b815d06751735e6a50fa46e913fdf85f751f0/hipamd/src/hip_platform.cpp#L362-L366
+#if defined(KOKKOS_ARCH_AMD_GFX1030) || defined(KOKKOS_ARCH_AMD_GFX1100) || \
+    defined(KOKKOS_ARCH_AMD_GFX1103)
```

Comment:
I'm guessing that this list will have to be keep appended as more supported architectures are added in the future.
Is it possible to form a condition similar to what was done in the link instead, by querying a device attribute to determine a lower WF count usage?

Reply:
Maybe but we have the same logic at other places, so you are not going to gain anything by doing that.

Reply:
Unless we expect the `const`-ness of `maxWavesPerCU` to be important for performance, I would prefer to see this query the HIP device properties as well. Looks like the `major` and `minor` fields of the device property struct are the same as this `device.isa().versionMajor()` in the commented link.

Reply:
> Unless we expect knowing maxWavesPerCU at compile time to be important for performance, I would prefer to see this query the HIP device properties as well.

We are not doing this for the pleasure of keeping tables in Kokkos. The peak and the sustainable values do not match. We care about the sustainable value not the peak one, that's why we hardcode the value.

Reply:
Sorry if I wasn't entirely clear earlier. I prefer a different method by which Kokkos selects which hard-coded `32` or `64` to use: I think we should query the HIP device properties rather than the Kokkos CMake configuration to choose between them.

The downside of querying the HIP device properties is that this hard-coded value becomes dynamic at run time (won't be `const int` since you'll switch at runtime between two hard-coded values). Unless this is expected to be a big performance problem, let's choose `32` or `64` based on the HIP device property `major` field, or some similar logic.